src/models/room.ts

```diff
@@ -933,6 +1051,17 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
             .getState(EventTimeline.BACKWARDS);
         this.currentState = this.getLiveTimeline()
             .getState(EventTimeline.FORWARDS);
+
+        // Let people know to register new listeners for the new state
+        // references. The reference won't necessarily change every time so only
+        // emit when we see a change.
+        if (previousOldState !== this.oldState) {
+            this.emit(RoomEvent.OldStateUpdated, this, previousOldState, this.oldState);
```

Comment:
`RoomEvent.OldStateUpdated` is not used but seemed applicable to add the equivalent to `RoomEvent.CurrentStateUpdated` since people could be doing the same thing with `oldState` for whatever reason. Also willing to remove since we aren't exercising it ourselves.